### ISSUE TITLE
404 page doesn't show with i18n and prefixDefaultLocale enabled

### ISSUE THREAD

**Issue by Alejandro-FA** [withastro/astro#12750]
### Astro Info

```block
Astro                    v5.0.6
Node                     v22.11.0
System                   macOS (arm64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/sitemap
```

### Describe the Bug

The bug is closely related to #12509, which was solved by #12525 when `prefixDefaultLocale` is set to `false`. However, the bug is still present when `prefixDefaultLocale` is set to `true`.

### Test setup

1. Open a new project and create localized folders, with a folder for the default locale too (i.e. `prefixDefaultLocale` set to `true`), as specified by the [i18n guide](https://docs.astro.build/en/guides/internationalization/).

2. Create a `404.astro` page in `/pages`.

### Results

Visiting any URL prefixed with one of the supported languages shows the custom 404 page.

Visiting any URL without a valid language prefix shows a blank page.

### Example

With `["es", "en", "fr"]` as supported locales and `en` as the default locale:

`/en/blabla` => 404 page
`/fr/blabla` => 404 page

`/blabla` => blank page
`/it/blabla` => blank page 

### What's the expected result?

Any path that doesn't exist should show the custom 404 page.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-leyrunqm?file=astro.config.mjs

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Alejandro-FA** [comment]
@ematipico Sorry to bother you, but I have seen that you solved the original bug some weeks ago. Are you able to reproduce this bug?

**Comment by ematipico** [comment]
Please refrain from tagging me, especially since I haven't interacted with the issue.

**Comment by ematipico** [comment]
Closing, I can't reproduce the bug from the reproduction. It works as expected

**Comment by Alejandro-FA** [comment]
Yes, I know that I should get a 404. However, instead of showing the 404 page defined in `pages/404.astro` it just shows a blank page. My understanding is that any page that it is not defined should show the 404 page under `pages/404.astro`. At least, this is what happens if `prefixDefaultLocale` is set to false.

**Comment by ematipico** [comment]
Sorry, using a path with a locale made things very confusing for me.

**Comment by Alejandro-FA** [comment]
My bad, I should have explained the issue more clearly.
